typhos/benchmark/device.py

```diff
@@ -0,0 +1,50 @@
+"""
+A generator of test devices.
+
+These are meant to be used for targetted benchmarking and typically are
+"extreme" in both size and in composition in order to make specific
+loading issues more obvious.
+
+Note that this currently only supports simple, flat devices.
+In the future it can be expanded to have complicated nesting devices,
+Kind information, different data types to test different widget types, etc.
+"""
+from ophyd.device import Device, Component as Cpt
+from ophyd.signal import Signal
+
+
+def make_test_device_class(name='TestClass', signal_class=Signal,
+                           include_prefix=False, num_signals=10):
+    """
+    Creates a test Device class.
+
+    Parameters
+    ----------
+    name: str, optional
```

Comment:
Docstring conventions as per new style guide

```suggestion
    name : str, optional
```

Reply:
Ah shoot, this habit is one that's ingrained pretty deeply for me. I'll go through and fix it everywhere.